src/sinks/aws_s_s/config.rs

```diff
@@ -88,13 +88,14 @@ pub enum Encoding {
 }
 
 #[derive(Clone, Debug)]
-pub struct ConfigWithIds {
-    pub base_config: BaseSSSinkConfig,
-    pub fifo: bool,
+pub struct MessageIdConfig {
+    message_group_id: Option<String>,
+    message_deduplication_id: Option<String>,
+    fifo: bool,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** Ah I should have made my suggestion a bit more clear, I apologize. What I had in mind was removing the need to have a struct like this altogether. So the impl functions of this struct turn into just standalone helper functions.

Something like this:

```Rust

// aws_s_s/config.rs

pub(super) fn message_group_id(message_group_id: Option<String>, fifo: bool)-> crate::Result<Option<Template>> { ... }

pub(super) fn message_deduplication_id(mesage_deduplication_id: Option<String>) -> crate::Result<Option<Template>> { ... }

// aws_s_s/request_builder.rs

impl SSRequestBuilder {
    pub fn new( encoding_config: EncodingConfig, message_group_id: Option<Template>, message_deduplication_id: Option<Template>) -> crate::Result<Self> { ... }

// sns/config.rs

    async fn build(
        &self,
        cx: SinkContext,
    ) -> crate::Result<(crate::sinks::VectorSink, crate::sinks::Healthcheck)> {

        // ...
        
        let request_builder = SSRequestBuilder::new(
            self.base_config.encoding.clone(),
            message_group_id(
                 self.base_config.message_group_id.clone(),
                 self.topic_arn.ends_with(".fifo")
            ),
             message_deduplication_id(self.base_config.message_deduplication_id.clone()
         )?;
         
        let sink = SSSink::new(
            request_builder,
            self.base_config.request.clone(),
            publisher,
        )?;
```

Reply:
done! Btw, @ArzelaAscoIi will take it from here! :-) Thanks for all of your help @neuronull!

Reply:
done :)